Flaky panic on `TestInit/sha256/With_Default_Branch`

### Bug Description

The test workflow action is failing intermittently due to a panic in `copyTreeToStorageRecursive` as per [this execution run](https://github.com/go-git/go-git/actions/runs/19524151748/job/55893360661):

```
--- FAIL: TestInit (0.00s)
    --- FAIL: TestInit/sha256 (0.00s)
        --- FAIL: TestInit/sha256/With_Default_Branch (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa5cabe]

goroutine 110664 [running]:
testing.tRunner.func1.2({0xc4f9a0, 0x4254ad0})
	/opt/hostedtoolcache/go/1.24.10/x64/src/testing/testing.go:1734 +0x21c
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.24.10/x64/src/testing/testing.go:1737 +0x35e
panic({0xc4f9a0?, 0x4254ad0?})
	/opt/hostedtoolcache/go/1.24.10/x64/src/runtime/panic.go:792 +0x132
github.com/go-git/go-git/v6.(*buildTreeHelper).copyTreeToStorageRecursive(0xc0007e7c88, {0xc0021af908, 0x7}, 0x0)
	/home/runner/work/go-git/go-git/worktree_commit.go:354 +0x7e
github.com/go-git/go-git/v6.(*buildTreeHelper).copyTreeToStorageRecursive(0xc0007e7c88, {0x0, 0x0}, 0xc00130a300)
	/home/runner/work/go-git/go-git/worktree_commit.go:363 +0x285
github.com/go-git/go-git/v6.(*buildTreeHelper).BuildTree(0xc0007e7c88, 0xc00017cd40, 0x481412?)
	/home/runner/work/go-git/go-git/worktree_commit.go:302 +0x20f
github.com/go-git/go-git/v6.(*Worktree).Commit(0xc000acaed0, {0xd31819, 0x13}, 0xc0007e7d78)
	/home/runner/work/go-git/go-git/worktree_commit.go:77 +0x2c5
github.com/go-git/go-git/v6.createCommit(0xc0001e6000, 0xc000acade0?)
	/home/runner/work/go-git/go-git/repository_common_test.go:54 +0x3d5
github.com/go-git/go-git/v6.TestInit.func1.1(0xc0001e6000)
	/home/runner/work/go-git/go-git/repository_test.go:92 +0x35b
testing.tRunner(0xc0001e6000, 0xc002626c30)
	/opt/hostedtoolcache/go/1.24.10/x64/src/testing/testing.go:1792 +0xf4
created by testing.(*T).Run in goroutine 110658
	/opt/hostedtoolcache/go/1.24.10/x64/src/testing/testing.go:1851 +0x413
FAIL	github.com/go-git/go-git/v6	57.432s
```

### go-git Version

f7f2b493a56ff81e9c7ee2637d4b1eb05fcf56c0 (as per execution run, however this is likely still happening in the `main` branch).

### Steps to Reproduce

_No response_

### Additional Information

_No response_

--- a/worktree_commit.go
+++ b/worktree_commit.go
@@ -355,7 +355,7 @@ func (se sortableEntries) Swap(i, j int)      { se[i], se[j] = se[j], se[i] }
 func (h *buildTreeHelper) copyTreeToStorageRecursive(parent string, t *object.Tree) (plumbing.Hash, error) {
 	sort.Sort(sortableEntries(t.Entries))
 	for i, e := range t.Entries {
-		if e.Mode != filemode.Dir && !e.Hash.IsZero() {
+		if e.Mode != filemode.Dir {
 			continue
 		}
 

--- a/worktree_commit_test.go
+++ b/worktree_commit_test.go
@@ -22,12 +22,42 @@ import (
 
 	"github.com/go-git/go-git/v6/plumbing"
 	"github.com/go-git/go-git/v6/plumbing/cache"
+	"github.com/go-git/go-git/v6/plumbing/filemode"
+	"github.com/go-git/go-git/v6/plumbing/format/index"
 	"github.com/go-git/go-git/v6/plumbing/object"
 	"github.com/go-git/go-git/v6/plumbing/storer"
 	"github.com/go-git/go-git/v6/plumbing/transport"
 	"github.com/go-git/go-git/v6/storage/filesystem"
 	"github.com/go-git/go-git/v6/storage/memory"
 )
+
+// This tests that buildTreeHelper correctly
+// handles file entries with zero hashes. This was a bug where files with
+// zero hashes would incorrectly trigger recursive tree building, causing
+// a nil pointer dereference panic.
+func TestBuildTreeHelper_1773(t *testing.T) {
+	fs := memfs.New()
+	storage := memory.NewStorage()
+
+	h := &buildTreeHelper{
+		fs: fs,
+		s:  storage,
+	}
+
+	// Create an index with a file that has a zero hash.
+	idx := &index.Index{
+		Entries: []*index.Entry{
+			{
+				Name: "file.txt",
+				Mode: filemode.Regular,
+				Hash: plumbing.ZeroHash,
+			},
+		},
+	}
+
+	_, err := h.BuildTree(idx, &CommitOptions{})
+	require.NoError(t, err)
+}
 
 func (s *WorktreeSuite) TestCommitEmptyOptions() {
 	fs := memfs.New()

